pool/src/lib.rs

```diff
@@ -65,6 +80,77 @@ impl LendingPoolTrait for LendingPool {
         Ok(())
     }
 
+    /// Configures the reserve collateralization parameters
+    /// all the values are expressed in percentages with two decimals of precision.
+    ///
+    /// # Arguments
+    ///
+    /// - asset The address of asset that should be set as collateral
+    /// - config Collateral parameters
+    ///
+    /// # Panics
+    ///
+    /// - Panics with `InvalidReserveParams` when wrong collateral params provided.
+    /// - Panics if the caller is not the admin.
+    fn configure_as_collateral(
+        env: Env,
+        asset: Address,
+        params: CollateralParams,
+    ) -> Result<(), Error> {
+        Self::ensure_admin(&env)?;
+
+        //validation of the parameters: the LTV can
+        //only be lower or equal than the liquidation threshold
+        //(otherwise a loan against the asset would cause instantaneous liquidation)
+        assert_with_error!(
+            &env,
+            params.ltv < params.liq_threshold,
+            Error::InvalidReserveParams
+        );
+
+        if params.liq_threshold != 0 {
+            //liquidation bonus must be bigger than 100.00%, otherwise the liquidator would receive less
+            //collateral than needed to cover the debt
+            assert_with_error!(
+                &env,
+                params.liq_bonus as i128 > PERCENTAGE_FACTOR,
+                Error::InvalidReserveParams
+            );
+
+            //if threshold * bonus is less than PERCENTAGE_FACTOR, it's guaranteed that at the moment
+            //a loan is taken there is enough collateral available to cover the liquidation bonus
+            assert_with_error!(
+                env,
+                params
+                    .liq_threshold
+                    .percent_mul(params.liq_bonus)
+                    .ok_or(Error::MathOverflowError)?
+                    <= PERCENTAGE_FACTOR,
+                Error::InvalidReserveParams
+            );
+        } else {
+            assert_with_error!(&env, params.liq_bonus == 0, Error::InvalidReserveParams);
+
+            //if the liquidation threshold is being set to 0,
```

Comment:
Should we check if `params.liq_threshold` is set to zero?
I see that we check if it's not zero above, but what if it's negative. I suppose it's possible if `params.ltv` is negative.

Reply:
It is not possible, because `params.ltv` has type `u32`